app/hotels/src/allHotels/index.js

```diff
@@ -0,0 +1,161 @@
+import { Layout, GeneralError } from '@kiwicom/react-native-app-shared';
+
+import AllHotelsSearchList from './AllHotelsSearchList';
+import { handleOpenSingleHotel } from '../singleHotel';
+import { sanitizeHotelFacilities } from '../GraphQLSanitizers';
+import { getSearchQueryParams } from '../search/SearchQueryHelpers';
+import SearchForm from './searchForm/SearchForm';
+import FilterStripe from '../filter/FilterStripe';
+import type { AvailableHotelSearchInput } from '../singleHotel/AvailableHotelSearchInput';
+import type { Coordinates } from '../CoordinatesType';
+import type {
+  SearchParams,
+  OnChangeSearchParams,
+} from './searchForm/SearchParametersType';
+import type {
+  FilterParams,
+  OnChangeFilterParams,
+} from '../filter/FilterParametersType';
+import type { allHotels_cityLookup_QueryResponse } from './__generated__/allHotels_cityLookup_Query.graphql';
+import type { allHotelsQueryResponse } from './__generated__/allHotelsQuery.graphql';
+
+type Props = {|
+  location: string,
+  search: SearchParams,
+  filter: FilterParams,
+  currency: string,
+  openSingleHotel: (searchParams: AvailableHotelSearchInput) => void,
+  onSearchChange: OnChangeSearchParams => void,
+  onFilterChange: OnChangeFilterParams => void,
+  onLocationChange: (location: string) => void,
+  onCityIdChange: (cityId: string | null) => void,
+  coordinates: Coordinates | null,
+|};
+
+export default class AllHotels extends React.Component<Props> {
+  componentDidMount = () => {
+    // submit form with the initial date interval parameters
+    this.props.onSearchChange({
+      checkin: moment()
+        .add(1, 'week')
+        .startOf('isoWeek')
+        .toDate(),
+      checkout: moment()
+        .add(1, 'week')
+        .endOf('isoWeek')
+        .toDate(),
+    });
+  };
+
+  handleOpenSingleHotel = (hotelId: string) => {
+    const { search, openSingleHotel } = this.props;
+    handleOpenSingleHotel(hotelId, search, openSingleHotel);
+  };
+
+  renderAllHotelsSearchList = (propsFromRenderer: allHotelsQueryResponse) => {
+    return (
+      <AllHotelsSearchList
+        data={propsFromRenderer}
+        openSingleHotel={this.handleOpenSingleHotel}
+        flatListHeader={
+          <View>
+            <SearchForm
+              onChange={this.props.onSearchChange}
+              onLocationChange={this.props.onLocationChange}
+              search={this.props.search}
+              location={this.props.location}
+            />
+            <FilterStripe
+              filter={this.props.filter}
+              onChange={this.props.onFilterChange}
+              currency={this.props.currency}
+            />
+          </View>
+        }
+      />
+    );
+  };
+
+  renderAllHotelsSearchPublicRenderer = (
+    rendererProps: allHotels_cityLookup_QueryResponse,
+  ) => {
+    const cityId = idx(rendererProps, _ => _.city.edges[0].node.id) || null;
+
+    // TODO: cannot query with non-existing cityId!
```

Comment:
Now typing in the location search bar closes the keyboard because component re-renders. I think you need to separate location from the SearchList, and rather just pass it the cityId, so re-render happens only when cityId changes.